go.mod

```diff
@@ -1,29 +1,29 @@
 module connectrpc.com/vanguard
 
-go 1.21
+go 1.22
```

Comment:
Shouldn't this be `1.22.0`?

Reply:
Why?

Reply:
In newer versions of Go this is supposed to be a full version number. In previous releases just two components was sufficient.

Reply:
It's not that it's _supposed_ to be a full version number -- simply that it is allowed to be. It's the `toolchain` directive that _must_ be a full version number. This one can still be a language level. I'd prefer we stick with the language level so that things "just work" for users that disable the automatic download of Go toolchains.

Reply:
Why are we upgrading to v1.22.0? Our general policy is to support three versions of Go. We should stick with 1.21.0, and downgrade whatever dep requires 1.22.0, until we revisit this policy and change it across the ConnectRPC org (and possibly across bufbuild org, too).

We likely will need to change to supporting just the past two versions (like official Go SDK support policy) due to things like this:
https://github.com/googleapis/google-cloud-go/commit/8a68e456d089e946bec93ffad6edb35eb0d29f93#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5